research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,213 @@
+| `min` | `min` | `number` | `0` | The minimum value of the slider. |
+
+#### Methods
+
+*Example Table*
+
+| Signature | Description |
+| ----------- | ---------- |
+| `increment(value: number = this.step): void` | Increments the `value` of the component by the amount specified by `step`, clamped within `min`/`max` values. |
+| `decrement(value: number = this.step): void` | Decrements the `value` of the component by the amount specified by `step`, clamped within `min`/`max` values. |
+
+#### Events
+
+*Example Table*
+
+| Event Name | Detail Type | Bubbles | Composed | Cancellable | Dispatch Behavior |
+| ---------- | ----------- | ------- | -------- | ----------- | ----------------- |
+| `change` | none | `true` | `true` | `false` | Fired when the slider's `value` changes. |
+
+### Appearance
+
+*Screenshots and/or description of the basic appearance of the component.*
+
+### Anatomy
+
+*Outline the component's structure with a diagram of its visual tree (shadow dom). Enumerate key areas of visual customization.*
+
+#### Diagram
+
+*Example*
+
+![Example Diagram](../images/spec-diagram-example.png)
+
+#### DOM Structure
+
+*Define the recommended DOM to represent the component's anatomy, as shown above. Show how important attributes (aria attributes especially) are applied to the various parts. In cases where a component nests other components, expand the full DOM structure to understand the expectation and any shortcomings in the child components' customizability.*
+
+#### Slots
+
+*Example Table*
+
+| Slot Name | Description | Fallback Content |
+| --------- | ----------- | ---------------- |
+| `thumb` | Enables replacing the component's `thumb` part with custom markup. | A default thumb implementation is provided. |
+
+#### Host Classes
+
+| Class Name | Description |
+| ---------- | ----------- |
+
+#### Slotted Content Classes
+
+| Class Name | Description |
+| ---------- | ----------- |
+
+#### CSS Parts
+
+*Example Table*
+
+| Part Name | Description |
+| --------- | ----------- |
+| `thumb` | Enables styling the `thumb` part without requiring it to be completely replaced via its slot. |
+
+---
+
+## Implementation Plan
+
+### States and Interactions
+
+*List key component states, valid state transitions, and how interactions trigger state transitions.*
+
+- *Are states changed as a result of keyboard? mouse? touch?*
+- *Are there corresponding keyboard interactions for mouse/touch interactions?*
+- *Are some events triggered on key down vs key press?*
+
+*Example Table*
+
+| State Group | States | Initial State | Description | Interaction/Transition |
+| ----------- | ------ | ------------- | ----------- | ---------------------- |
+| `disabled` | `true`/`false` | `false` | When `true`, disables the control, preventing user interaction and displaying the control with a disabled appearance. | No interaction. Controlled programmatically. |
+
+### Accessibility
+
+*Consider the accessibility of the component.*
```

Comment:
This is very vague. Accessibility is a combo of: screen reader compatibility, keyboard model, contrast ratios, avoiding color-only to indicate some state (focus rectangles, for example).

Should this be specific to screen reader concerns and which aria attributes need to be distributed across which elements?

Reply:
I like the ambiguity here personally. By going broad here I think someone needs to really internalize and consider what all is necessary to include here. For instance, for something like a dialog (which I plan on using this template for) it may be enough to link to the W3C aria-practices documentation as a pattern already exists in standards form. For something like a split-button though, that doesn't exist and I would expect much more thought to go into defining how that might need to be built given the existing standards.

Reply:
I was treating that as a broad overview statement with the sections below on keyboard nav, focus, forms, and AT tech being the areas where more details were fleshed out. We can add something maybe about how states would be represented in a color-independent way. I'll leave this section as is for now and we can discuss f2f as part of a broader review.

Reply:
Yep I retract my statement - this is more of a group of accessibility concerns organized by the subtitles below :)